Cannot build on 32-bit architectures

As of https://github.com/wyyerd/pulsar-rs/commit/fb04666dc6522452ecea372f42e24e0b4c068119 it looks like this crate cannot be build on 32 bit architectures due to:

```
error[E0277]: the trait bound `u64: ToUsize` is not satisfied
Error:    --> /cargo/registry/src/github.com-1ecc6299db9ec823/pulsar-4.1.0/src/message.rs:500:29
    |
500 |     let (i, payload) = take(metadata.payload_size as u64)(i)?;
    |                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `ToUsize` is not implemented for `u64`
    | 
   ::: /cargo/registry/src/github.com-1ecc6299db9ec823/nom-7.0.0/src/bytes/streaming.rs:426:6
    |
426 |   C: ToUsize,
    |      ------- required by this bound in `nom::bytes::streaming::take`
```

Given the comment higher up it seems like `payload_size` is actually a i32 and so should fit within a u32?

Can be tested via `cross`:

```
cross build --target armv7-unknown-linux-musleabihf
```

I'll open a PR shortly.

--- a/src/message.rs
+++ b/src/message.rs
@@ -497,7 +497,7 @@ fn batched_message(i: &[u8]) -> IResult<&[u8], BatchedMessage> {
         |metadata| metadata.payload_size >= 0,
     )(i)?;
 
-    let (i, payload) = take(metadata.payload_size as u64)(i)?;
+    let (i, payload) = take(metadata.payload_size as u32)(i)?;
 
     Ok((
         i,

